lib/spack/spack/binary_distribution.py

```diff
@@ -1119,7 +1121,6 @@ def relocate_package(spec, allow_root):
     prefix_to_prefix_bin = OrderedDict({})
 
     if old_sbang_install_path:
-        import spack.hooks.sbang as sbang
```

Comment:
There was some kind of circular import issue on certain platforms that led to this being nested here.  Can you look at the blame for these nested `sbang` imports and see if you can figure out whether it's really been resolved?  I think @eugeneswalker and @scheibelp should comment.

Reply:
The imports are still nested, but they are done at the top of the function. This is because imports act like a local assignment to the `spack` variable and using any `spack.*` module or function before the nested import will fail with weird errors. Checking `git blame` right now.

Reply:
Only reference found for these imports https://github.com/spack/spack/pull/20768#discussion_r555265884